src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const group: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = `${item[key]}`;
+
+    if (!group[groupKey]) {
```

Comment:
You've done the same condition, but in the shortened form) You should use `hadOwnPropery()` in such case. But okay, we'll omit this)